src/OpenTelemetry.Api/Trace/TracerProvider.cs

```diff
@@ -50,5 +55,67 @@ public Tracer GetTracer(string name, string version = null)
 
             return new Tracer(new ActivitySource(name, version));
         }
+
+        /// <summary>
+        /// Attempts to shutdown the processor, blocks the current thread until
+        /// shutdown completed or timed out.
+        /// </summary>
+        /// <param name="timeoutMilliseconds">
+        /// The number of milliseconds to wait, or <c>Timeout.Infinite</c> to
+        /// wait indefinitely.
+        /// </param>
+        /// <returns>
+        /// Returns <c>true</c> when shutdown succeeded; otherwise, <c>false</c>.
+        /// </returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when the <c>timeoutMilliseconds</c> is smaller than -1.
+        /// </exception>
+        /// <remarks>
+        /// This function guarantees thread-safety. Only the first call will
+        /// win, subsequent calls will be no-op.
+        /// </remarks>
+        public bool Shutdown(int timeoutMilliseconds = Timeout.Infinite)
```

Comment:
On reading the spec again and the usage of this, I am doubting if this should be in the API...I mean why would allow libraries which depend on this API to call shutdown? 
It should be a pure SDK concern. We can implement it as Extension method on TracerProvider, similar to AddProcessor. That ensures that you can call shutdown *only* if you are the application owner who creates the TracerProvider.

Reply:
@cijothomas I could remove it from API if we don't want to allow the API only users to be able to call shutdown on it.

We could just have it implemented in the TracerProviderSDK as an instance method then. It already has AddProcessor as an instance method. Why should we make it an extension method?

Reply:
The instance method is internal and the extension method is public?

Reply:
Ahh yes!

Reply:
yes.
There is no need of an API user, (who is just describing the "shape" of traces) to shutdown a provider. Hence the ask to move from api to sdk.

Reply:
If we move this to TracerProviderExtensions, we would still be exposing this to the API just like how the AddProcessorExtension method under TracerProviderExtensions allows the API to call AddProcessor. We could always make the Shutdown extension method a no-op for API but we would still be exposing it.

The extension methods take in TracerProvider as the parameter as that's the public class. We cannot use TracerProviderSdk as it's internal. We could create a new public class which the TracerProviderSdk derives from to restrict the Shutdown method to just SDK.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/1507 Check if this PR makes sense. If yes, we can borrow that approach.

Reply:
This would just change it to an extension method from an instance method, right? It'll just be no-op for API which the OnShutdown instance method of TracerProvider ensured as well. It's definitely cleaner to make that an extension method but  why would we want to expose AddProcessor and Shutdown even as extension methods to API? I didn't completely understand this from the original comment: _"That ensures that you can call shutdown only if you are the application owner who creates the TracerProvider."_

Reply:
Its not very clean either. But atleast it ensures that a library author who accesses the Default TracerProvider, will not be able to force its shutdown!

Reply:
Fixed this.